src/Microsoft.AspNetCore.DataProtection/TypeForwardingActivator.cs

```diff
@@ -39,16 +40,14 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
                 forwardedTypeName = originalTypeName.Replace(OldNamespace, CurrentNamespace);
             }
 
-#if NET46
-            if (candidate || forwardedTypeName.Contains(CurrentNamespace))
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
```

Comment:
> Is there an actual reliable way to check the platform that doesn't use strings?

We could continue cross-compiling the binary. I filed an ask in CoreFx to see if they can expose a first class API: https://github.com/dotnet/corefx/issues/19927. They use FrameworkDescription in test code, but it doesn't seem reliable enough for shipping code especially given that we're tied to the `RuntimeInformation` reference and not to it's implementation.